src/test/java/org/sagebionetworks/bridge/spring/util/EtagComponentTest.java

```diff
@@ -264,4 +298,112 @@ public void nullBridgeSessionHeader() throws Throwable {
+
+        when(mockRequest.getHeader(IF_NONE_MATCH)).thenReturn(ETAG);
+        
+        component.checkEtag(mockJoinPoint);
+        
+        verify(mockCacheProvider, never()).removeObject(any());
+    }
+
+    @Test
+    public void invalidateCacheOnChange_valuesHaveNotChanged() throws Throwable {
+        when(mockSession.getParticipant()).thenReturn(new StudyParticipant.Builder()
+                .withClientTimeZone(TEST_CLIENT_TIME_ZONE).build());
+        
+        doReturn(ImmutableList.of(TIME_ZONE_ANN)).when(mockContext).getCacheKeys();
+        doReturn(ImmutableMap.of("clientTimeZone", TEST_CLIENT_TIME_ZONE)).when(mockContext).getArgValues();
+
+        when(mockRequest.getHeader(IF_NONE_MATCH)).thenReturn(ETAG);
+        
+        component.checkEtag(mockJoinPoint);
+        
+        verify(mockCacheProvider, never()).removeObject(any());
+    }
+
+    @Test
+    public void invalidateCacheOnChange_newValue() throws Throwable {
+        // this is change, because there's no value in the session at this point
+        
+        doReturn(ImmutableList.of(TIME_ZONE_ANN)).when(mockContext).getCacheKeys();
+        doReturn(ImmutableMap.of("userId", TEST_USER_ID, "clientTimeZone", TEST_CLIENT_TIME_ZONE))
+            .when(mockContext).getArgValues();
+
+        when(mockRequest.getHeader(IF_NONE_MATCH)).thenReturn(ETAG);
+        
+        when(mockCacheProvider.getObject(
+                CacheKey.etag(DateTimeZone.class, TEST_USER_ID), DateTime.class)).thenReturn(MODIFIED_ON);
+        when(mockRequest.getHeader(IF_NONE_MATCH)).thenReturn(ETAG);
+        
+        when(mockMd5DigestUtils.digest(MODIFIED_ON.toString().getBytes())).thenReturn("ETAG".getBytes());
+        
+        component.checkEtag(mockJoinPoint);
+        
+        verify(mockCacheProvider).removeObject(CacheKey.etag(DateTimeZone.class, TEST_USER_ID));
+        verify(mockMd5DigestUtils).digest((byte[])any());
+    }
+
+    @Test
+    public void invalidateCacheOnChange_changedValue() throws Throwable {
+        when(mockSession.getParticipant()).thenReturn(new StudyParticipant.Builder()
+                .withClientTimeZone("America/Chicago").build());
+        
+        doReturn(ImmutableList.of(TIME_ZONE_ANN)).when(mockContext).getCacheKeys();
+        doReturn(ImmutableMap.of("userId", TEST_USER_ID, "clientTimeZone", TEST_CLIENT_TIME_ZONE))
+            .when(mockContext).getArgValues();
+        
+        when(mockCacheProvider.getObject(
+                CacheKey.etag(DateTimeZone.class, TEST_USER_ID), DateTime.class)).thenReturn(MODIFIED_ON);
+        when(mockRequest.getHeader(IF_NONE_MATCH)).thenReturn(ETAG);
+        
+        when(mockMd5DigestUtils.digest(MODIFIED_ON.toString().getBytes())).thenReturn("ETAG".getBytes());
+        
+        component.checkEtag(mockJoinPoint);
+        
+        verify(mockCacheProvider).removeObject(CacheKey.etag(DateTimeZone.class, TEST_USER_ID));
+        verify(mockMd5DigestUtils).digest((byte[])any());
```

Comment:
so does this run twice, it would have to recalculate the tag...in which case, isn't it the same case as below?

Reply:
Added some notes on why these are different, and a couple more assertions to each.